services/ghibliService.go

```diff
@@ -84,6 +93,54 @@ func filterFilmsById(films [][]string, id string) (models.GhibliFilm, error) {
 	return models.GhibliFilm{}, fmt.Errorf("film with id %s not found in repository", id)
 }
 
+// GetFilm retieves a record from the repository
+func (gs *ghibliService) GetFilmsConcurrently(query url.Values) ([]models.GhibliFilm, error) {
+	typeParam := query.Get("type")
+	maxItemsParam := query.Get("items")
+	maxItemsPerWorkerParam := query.Get("items_per_workers")
+
+	films, err := gs.filsmRepo.ReadCSVFile()
+	if err != nil {
+		log.Printf("Failed to fetch film from repository: %s", err)
+		return []models.GhibliFilm{}, err
+	}
+	var jobs []*workerspool.Job
+	maxItemsPerWorkerNum, _ := strconv.Atoi(maxItemsPerWorkerParam)
+
+	for _, film := range films {
+		jobs = append(jobs, workerspool.NewJob(workerProcess, workerParams{
+			Type:     typeParam,
+			MaxItems: maxItemsPerWorkerParam,
+			Record:   film,
+		}, maxItemsPerWorkerNum))
+	}
+
+	maxItemsParamNum, _ := strconv.Atoi(maxItemsParam)
+	numWorkers := maxItemsParamNum/2 - 1
+	if numWorkers <= 0 {
+		numWorkers = 1
+	}
+
+	pool := workerspool.NewWorkersPool(jobs, numWorkers)
+	pool.Run()
```

Comment:
You will wait for your pool to finish this, in this case why you did not retrieve your results from pool.Results as this should contain concurrency results.